scintillometry/fourier/base.py

```diff
@@ -0,0 +1,273 @@
+            ``axes[-1]``.
+        norm : 'ortho' or None, optional
+            If `None` (default), uses an unscaled forward transform and 1 / n
+            scaled inverse transform.  If 'ortho', uses a 1 / sqrt(n) scaling
+            for both.
+        verify : bool, optional
+            Verify setup is successful and self-consistent.
+        """
+
+        # Extract information if user passed in a dummy array.
+        if isinstance(input_data, np.ndarray):
+            input_data = {'shape': input_data.shape,
+                          'dtype': input_data.dtype}
+        elif isinstance(input_data['dtype'], str):
+            input_data['dtype'] = np.dtype(input_data['dtype'])
+
+        # Set direction, axis and normalization.  If axis is None, set it to 0.
+        direction = direction if direction == 'inverse' else 'forward'
+        axis = operator.index(axis)
+        norm = norm if norm == 'ortho' else None
+
+        # Determine frequency-domain shape.
+        output_shape = list(input_data['shape'])
+        # The inverse of a real-valued transform should return real values.
+        if direction == 'inverse':
+            assert 'complex' in input_data['dtype'].name, (
+                "frequency-domain array must be complex.")
+            if real_transform:
+                output_shape[axis] = output_shape[axis] * 2
+                output_dtype = np.dtype('float{0:d}'.format(
+                    input_data['dtype'].itemsize * 8 // 2))
+            else:
+                output_dtype = input_data['dtype']
+
+            freq_data = input_data
+            time_data = {'shape': tuple(output_shape),
+                         'dtype': output_dtype}
+        else:
+            if 'float' in input_data['dtype'].name:
+                # assert not (input_data['shape'][axis] % 2), (
+                #     "time-domain array must have an even number of elements "
+                #     "along the axis being transformed.")
+                output_shape[axis] = output_shape[axis] // 2 + 1
+                output_dtype = np.dtype('complex{0:d}'.format(
+                    input_data['dtype'].itemsize * 8 * 2))
+            else:
+                output_dtype = input_data['dtype']
+
+            time_data = input_data
+            freq_data = {'shape': tuple(output_shape),
+                         'dtype': output_dtype}
+
+        # Store time and frequency-domain array shapes.
+        data_format = {'time_shape': time_data['shape'],
+                       'time_dtype': time_data['dtype'],
+                       'freq_shape': freq_data['shape'],
+                       'freq_dtype': freq_data['dtype']}
+
+        return self._setup_transform(data_format, direction, axis, norm,
+                                     sample_rate)
+
+    def _setup_transform(data_format, direction, axis, norm, sample_rate):
+        raise NotImplementedError()
+
+    def inverse(self, data_format, direction, axis, norm, sample_rate):
```

Comment:
I don't understand the point of this; or, at least not unless it would just take a FFT instance as its sole argument.